code/graph_algorithms/src/best_first_search/best_first_search.cpp

```diff
@@ -0,0 +1,83 @@
+#include <iostream>
+#include <map>
+#include <set>
+#include <queue>
+
+/* Part of Cosmos by OpenGenus Foundation */
+
+
+void bestFirstSearch(int start,                                             // the node that we are currently at
+                     int goal,                                              // the node that we are looking for
+                     std::map<int, std::set<std::pair<int, int>>>& adjList, // the adjacency list for each node (cost, nodeNumber)
+                     std::map<int, bool>& visited)                          // the 'visited' state for each node
+{
+
+    // This priority queue wil hold the node values, ordered by costs
+    std::priority_queue<std::pair<int, int>, std::vector<std::pair<int, int>>, std::greater<std::pair<int, int>>> Q;
+
+    // Add the first node with cost 0 to the queue and mark it as visited
+    Q.push(std::make_pair(0, start));
+    visited[start] = true;
+
+    // Repeat the algorithm until there are no more nodes left to process
+    while (!Q.empty()){
+        // Take the node width the minimum cost in the queue and then pop it off the queue
+        int current = Q.top().second;
+        Q.pop();
+
+        std::cout<<current<<" ";
+
+        // If this is the node that we are looking for, exit
+        if (current == goal){
+            return;
```

Comment:
Not clear on the intention. What about the remaining nodes in the queue? Do you need a priority queue. Nothing being done on the cost.